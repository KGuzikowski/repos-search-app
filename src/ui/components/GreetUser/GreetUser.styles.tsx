import styled from 'styled-components'
import { secondaryColor } from '../../style-variables'

export const GreetDiv = styled.div`
    width: 100%;
    margin: 0px 0px 40px 0;
    display: flex;
    flex-direction: column;
    align-items: center;

    @media only screen and (max-width: 700px) {
        margin: 0 0 20px 0;
    }
`

export const Greet = styled.p`
    font-weight: bold;
    color: ${secondaryColor};
    font-size: 30px;

    @media only screen and (max-width: 700px) {
        font-size: 25px;
    }
`

export const Info = styled.p`
    margin-top: 10px;
    font-size: 19px;

    @media only screen and (max-width: 700px) {
        font-size: 15px;
        margin-top: 5px;
    }
`