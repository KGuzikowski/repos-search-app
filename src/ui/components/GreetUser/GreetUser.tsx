import React from 'react'
import { GreetDiv, Greet, Info } from './GreetUser.styles'

interface GreetUserType {
    user: string
}

const GreetUser = ({ user }: GreetUserType) => {
    return (
        <GreetDiv>
            <Greet>
                Hi { user }!
            </Greet>
            <Info>
                (Your repositories will be highlighted)
            </Info>
        </GreetDiv>
    )
}

export default GreetUser