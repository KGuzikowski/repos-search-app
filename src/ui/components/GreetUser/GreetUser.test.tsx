import React from 'react'
import { create } from 'react-test-renderer'
import GreetUser from './GreetUser'

it('renders GreetUser component correctly', () => {
    const component = create(<GreetUser user='Karol' />)
    expect(component.toJSON()).toMatchSnapshot()
})