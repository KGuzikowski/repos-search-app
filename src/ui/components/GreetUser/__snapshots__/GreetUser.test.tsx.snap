// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GreetUser component correctly 1`] = `
<div
  className="sc-AxjAm jQNnlu"
>
  <p
    className="sc-AxirZ isHfjn"
  >
    Hi 
    Karol
    !
  </p>
  <p
    className="sc-AxiKw fimrdQ"
  >
    (Your repositories will be highlighted)
  </p>
</div>
`;
